syntax = "proto3";


// Calculator service definition
service Calculator {
  // Makes calculation on given numbers
  rpc Calc (CalcRequest) returns (CalcReply) {}
}

// Request message containing two numbers and operation to calculate
message CalcRequest {
  sint32 number_1 = 1;
  sint32 number_2 = 2;
  string operation = 3;
}

// Response message containing calculation result
message CalcReply {
  string res = 1;
}